app/models/gem_dependent.rb

```diff
@@ -0,0 +1,50 @@
+class GemDependent
+  attr_reader :gem_names
+
+  def initialize(gem_names)
+    @gem_names = gem_names
+  end
+
+  def memcached_client
+    Rails.cache
+  end
+
+  def fetch_dependencies
+    dependencies = []
+    keys = @gem_names.map { |g| "deps/v1/#{g}" }
+
+    memcached_gem_info = memcached_client.read_multi(*keys)
+
+    keys.each_with_index do |key, _i|
+      next unless memcached_gem_info.include?(key)
+      StatsD.increment 'gem_dependent.memcached.hit'
+      keys.delete(key)
+      dependencies << memcached_gem_info[key]
+    end
+
+    keys.each do |gem_key|
+      StatsD.increment 'gem_dependent.memcached.miss'
+      gem_name = gem_key.gsub("deps/v1/", "")
+      result = fetch_dependency_from_db(gem_name)
+      memcached_client.write(gem_key, result)
+      dependencies << result
+    end
+
+    dependencies
+  end
+
+  alias to_a fetch_dependencies
+
+  def fetch_dependency_from_db(gem_name)
+    gem_record = Rubygem.includes(:versions).find_by_name!(gem_name)
+    gem_record.versions.sort_by(&:number).reverse_each.map do |version|
+      version_deps = version.dependencies.find_all { |d| d.scope == "runtime" }
```

Comment:
`versions.sort_by(&:number)` should be `versions.includes(:dependencies).sort_by(&:number)` I think.